src/ui_plugin/systems/search.rs

```diff
@@ -91,6 +95,12 @@ pub fn search_box_text_changed(
                 let result = fuzzy_search(index, str.as_str());
                 match result {
                     Ok(docs) => {
+                        let mut doc_ids = docs
```

Comment:
node_ids

Reply:
Not sure if it worth it but viewport can only have nodes from current document, so we could filter the nodes by that, current document can be obtained in AppState.